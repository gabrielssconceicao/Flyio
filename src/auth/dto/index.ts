import { LoginDto } from './login.dto';
import { LoginResponseDto } from './login-response.dto';
import { RefreshTokenDto } from './refresh-token.dto';
import { TokenPayloadDto } from './token-payload.dto';

export { LoginDto, LoginResponseDto, RefreshTokenDto, TokenPayloadDto };
