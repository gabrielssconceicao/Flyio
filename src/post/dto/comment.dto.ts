import { ApiProperty } from '@nestjs/swagger';

class UserComment {
  @ApiProperty({ example: 'John Doe', description: 'User name' })
  name: string;

  @ApiProperty({ example: 'jdoe', description: 'User username' })
  username: string;

  @ApiProperty({
    example: 'https://example.com/profile.jpg',
    description: 'User profile image URL',
  })
  profileImg: string | null;
}

export class CommentDto {
  @ApiProperty({ example: 'd-43-5df-df4', description: 'Comment identifier' })
  id: string;

  @ApiProperty({ example: 'Hello world', description: 'Comment content' })
  content: string;

  @ApiProperty({ example: '2023-03-15 14:30:00', description: 'Comment date' })
  createdAt: Date;

  @ApiProperty({ type: UserComment, description: 'Comment creator' })
  user: UserComment;
}
