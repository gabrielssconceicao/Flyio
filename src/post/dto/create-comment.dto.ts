import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString, MaxLength, MinLength } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';

export class CreateCommentDto {
  @ApiProperty({
    example: 'Hello world',
    description: 'Comment text',
    minLength: 1,
    maxLength: 255,
  })
  @IsNotEmpty({ message: dtoErrorMessages('Content', DtoErrorType.REQUIRED) })
  @IsString({ message: dtoErrorMessages('Content', DtoErrorType.IS_STRING) })
  @MinLength(1, {
    message: dtoErrorMessages('Content', DtoErrorType.MIN_LENGTH, 1),
  })
  @MaxLength(255, {
    message: dtoErrorMessages('Content', DtoErrorType.MAX_LENGTH, 255),
  })
  content: string;
}
