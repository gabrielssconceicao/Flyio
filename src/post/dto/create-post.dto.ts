import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString, MinLength, MaxLength } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';

export class CreatePostDto {
  @ApiProperty({
    example: 'Hello world',
    description: 'Post text',
    minLength: 1,
    maxLength: 255,
  })
  @IsNotEmpty({ message: dtoErrorMessages('Text', DtoErrorType.REQUIRED) })
  @IsString({ message: dtoErrorMessages('Text', DtoErrorType.IS_STRING) })
  @MinLength(1, {
    message: dtoErrorMessages('Text', DtoErrorType.MIN_LENGTH, 1),
  })
  @MaxLength(255, {
    message: dtoErrorMessages('Text', DtoErrorType.MAX_LENGTH, 255),
  })
  text: string;
}
