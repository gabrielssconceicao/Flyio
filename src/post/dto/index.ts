import { CreatePostDto } from './create-post.dto';
import { FindAllPostsResponseDto } from './find-all-posts.dto';
import { FindAllLikedPostsResponseDto } from './find-all-liked-post.dto';
export { CreatePostDto, FindAllPostsResponseDto, FindAllLikedPostsResponseDto };
