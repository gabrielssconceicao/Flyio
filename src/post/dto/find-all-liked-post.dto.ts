import { ApiProperty } from '@nestjs/swagger';
import { PostEntity } from '../entities/post.entity';
import { FindAllResponseDto } from '../../common/dto/find-all-response.dto';

export class FindAllLikedPostsResponseDto extends FindAllResponseDto<PostEntity> {
  @ApiProperty({
    type: [PostEntity],
    description: 'List of posts liked by the user',
  })
  items: PostEntity[];
}
