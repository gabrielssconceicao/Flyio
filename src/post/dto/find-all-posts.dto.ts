import { ApiProperty } from '@nestjs/swagger';
import { PostEntity } from '../entities/post.entity';
import { FindAllResponseDto } from '../../common/dto/find-all-response.dto';

export class FindAllUsersResponseDto extends FindAllResponseDto<PostEntity> {
  @ApiProperty({
    type: [PostEntity],
    description: 'List of posts of all users',
  })
  items: PostEntity[];
}
