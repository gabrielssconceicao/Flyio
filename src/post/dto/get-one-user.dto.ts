import { ApiProperty } from '@nestjs/swagger';
import { PostEntity } from '../entities/post.entity';
import { CommentDto } from './comment.dto';

export class GetOnePostDto extends PostEntity {
  @ApiProperty({
    isArray: true,
    description: 'List of comments',
  })
  comments: CommentDto[];
}
