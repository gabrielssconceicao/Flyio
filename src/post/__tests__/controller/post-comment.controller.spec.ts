import { Test, TestingModule } from '@nestjs/testing';
import { PostCommentController } from '../../post-comment.controller';
import { PostCommentService } from '../../post-comment.service';
import { CommentDto } from '../../dto';
import { TokenPayloadDto } from 'src/auth/dto';
import {
  generateTokenPayloadDtoMock,
  jwtConfigurationMock,
  jwtServiceMock,
} from 'src/auth/mocks';
import { JwtService } from '@nestjs/jwt';
import jwtConfig from 'src/auth/config/jwt.config';

describe('PostController', () => {
  let controller: PostCommentController;
  let service: PostCommentService;

  let tokenPayload: TokenPayloadDto;
  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      controllers: [PostCommentController],
      providers: [
        {
          provide: PostCommentService,
          useValue: {
            create: jest.fn(),
          },
        },
        {
          provide: JwtService,
          useValue: jwtServiceMock,
        },
        {
          provide: jwtConfig.KEY,
          useValue: jwtConfigurationMock,
        },
      ],
    }).compile();

    controller = module.get<PostCommentController>(PostCommentController);
    service = module.get<PostCommentService>(PostCommentService);

    tokenPayload = generateTokenPayloadDtoMock();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be defined', () => {
    expect(controller).toBeDefined();
    expect(service).toBeDefined();
  });

  describe('<Create />', () => {
    it('should create a comment successfully', async () => {
      const commentDto: CommentDto = {
        id: '1',
        createdAt: new Date('2023-03-15 14:30:00'),
        content: 'test',
        user: { username: 'test', profileImg: null, name: 'test' },
      };
      const createCommentDto = { content: 'test' };
      jest.spyOn(service, 'create').mockResolvedValue(commentDto);

      const result = await controller.create(
        '1',
        createCommentDto,
        tokenPayload,
      );

      expect(service.create).toHaveBeenCalledWith(
        '1',
        createCommentDto,
        tokenPayload,
      );
      expect(result).toEqual(commentDto);
      expect(result).toMatchSnapshot();
    });
  });

  describe('<Delete />', () => {
    it('should delete a comment successfully', async () => {
      const commentId = '1';
      jest.spyOn(service, 'delete').mockResolvedValue();
      await controller.delete(commentId, tokenPayload);
      expect(service.delete).toHaveBeenCalledWith(commentId, tokenPayload);
    });
  });
});
