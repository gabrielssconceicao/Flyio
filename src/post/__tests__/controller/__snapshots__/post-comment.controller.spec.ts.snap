// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostController <Create /> should create a comment successfully 1`] = `
{
  "content": "test",
  "createdAt": 2023-03-15T17:30:00.000Z,
  "id": "1",
  "user": {
    "name": "test",
    "profileImg": null,
    "username": "test",
  },
}
`;
